models/classes/class.QtiTestService.php

```diff
@@ -882,17 +882,24 @@ private function saveDoc( core_kernel_classes_Resource $test, XmlDocument $doc){
      */
     public function createContent( core_kernel_classes_Resource $test, $createTestFile = true) {
 
-    	$props = self::getQtiTestDirectory()->getPropertiesValues(array(
-				PROPERTY_FILE_FILESYSTEM,
-				PROPERTY_FILE_FILEPATH
-			));
+        $dir = $this->getTestFile($test);
+        if (is_null($dir)) {
+            $props = self::getQtiTestDirectory()->getPropertiesValues(array(
+                    PROPERTY_FILE_FILESYSTEM,
+                    PROPERTY_FILE_FILEPATH
+                ));
+
+            $repository = new core_kernel_versioning_Repository(current($props[PROPERTY_FILE_FILESYSTEM]));
+            $path = (string) current($props[PROPERTY_FILE_FILEPATH]);
 
-        $repository = new core_kernel_versioning_Repository(current($props[PROPERTY_FILE_FILESYSTEM]));
-        $path = (string) current($props[PROPERTY_FILE_FILEPATH]);
+            // $directory is the directory where test related resources will be stored.
+            $directory = $repository->createFile('', $path .DIRECTORY_SEPARATOR. md5($test->getUri()) . DIRECTORY_SEPARATOR);
+            $dirPath = $directory->getAbsolutePath().DIRECTORY_SEPARATOR;
+        } else {
+            $directory = new core_kernel_file_File($dir);
```

Comment:
according to ::getTestFile signature, $dir is core_kernel_file_File already and there is no need for recasting